src/main/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitChecker.java

```diff
@@ -4,17 +4,16 @@
 import hudson.Util;
 import hudson.model.TaskListener;
 import hudson.util.LogTaskListener;
-import org.jenkinsci.plugins.github.config.GitHubServerConfig;
-import org.kohsuke.github.GHRateLimit;
-import org.kohsuke.github.GitHub;
-import org.kohsuke.github.RateLimitChecker;
-
 import java.io.IOException;
 import java.util.Objects;
 import java.util.Random;
 import java.util.concurrent.TimeUnit;
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import org.jenkinsci.plugins.github.config.GitHubServerConfig;
+import org.kohsuke.github.GHRateLimit;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.github.RateLimitChecker;
```

Comment:
Switched to a simple alphabetical order IIUC? Very welcome. Humans should not be reading `import` statements to begin with; normalizing the order minimizes the chance of stupid merge conflicts.

Reply:
@jglick  
It is my understanding that what code style guidelines there are for Jenkins say "Simple alphabetical non-static, then simple alphabetical static".   Is that incorrect? 

@jtnord 
What you prefer the ordering be?  I don't care what it is as long as I never have to tell someone in a code review to change it.

Reply:
So long as it is simple I would not really care. (Whether `static` `import`s should be allowed at all, and under what conditions, is a contentious topic.)

Reply:
@jglick 
The topic of `static` `import`s will not be solved by a formatter.

Reply:
>What you prefer the ordering be? I don't care what it is as long as I never have to tell someone in a code review to change it.

I have personal preferences but I do not care too much about the order apart from always non static before static imports though, and static imports should be reserved for test code apart from those very rare special occasions.

But given as you asked, my preference is someone along the following lines.

1. core libraries first (`java`, `javax`)
2. then 3rd party libraries (`com` .`edu` , `org`)
3. then a break 
4. then jenkins core code (`hudson`, `jenkins`, `org.jenkinsci`, `io.jenkins`)
5. then stuff from my own project (e.g. `org.jenkinsci.plugins.github_branch_source`)
6. then a break
7. then the same but for static imports.

That said my IDE will automatically an import put it where it is configured to by its ordering (eclipse) and having per project settings is a PITA (I do not recall the last time I typed an import statement).  I try to reduce diff bloat it after the fact by manually moving it if I recall :)


> The topic of static imports will not be solved by a formatter.

you mean the ordering or that they should or should not be allowed?

Reply:
Whether or not they should be allowed.

Reply:
FWIW my personal preference is strictly alphabetical, no line breaks, no attempt to group in any way other than `| sort`.